src/lib/hooks/use-plan-filter.ts

```
@@ -13,50 +11,61 @@ import { useDeletePlanMutation } from '@/lib/mutations/use-delete-plan-mutation'
    */
-  const resetFilter = () => {
-    setFilter({
-      type: FILTER_TYPES.PUBLIC,
-      value: PUBLIC_OPTIONS.ALL,
+  const removeFilter = (type: FilterType) => {
+    setFilters((prev) => {
+      const newFilters = { ...prev };
+      if (type === FILTER_TYPES.DATE) delete newFilters.date;
+      if (type === FILTER_TYPES.KEYWORD) delete newFilters.keyword;
+      if (type === FILTER_TYPES.PUBLIC) delete newFilters.public;
+      return newFilters;
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)